pkg/client/userd/trafficmgr/traffic_manager.go

```
@@ -351,6 +351,29 @@ func connectCluster(c context.Context, cr *rpc.ConnectRequest) (*k8s.Cluster, er
 	return cluster, nil
 }
 
+func EnsureManager(ctx context.Context, req *rpc.InstallRequest) error {
+	// seg guard
+	cr := req.GetConnectRequest()
+	if cr == nil {
+		dlog.Warn(ctx, "Connect_request in install_request was nil, using defaults")
```

What is the user intended to do when seeing this warning in the log? I.e. what's the warning actually about?

---

A user should not see this, as it should never happen when using the `helm install` command. I thought it would be good practice to prevent panic.

---

They way I see it, it's either acceptable that the cr is `nil` at this point (no warning), or it isn't, in which case the function should return an error. Logging a warning and assigning a zero value covers a use-case that I don't think we want to have, so it just adds confusion when looking at the code.

---

> They way I see it, it's either acceptable that the cr is nil at this point (no warning), or it isn't

Seems safer to have it be acceptable. I can change the `warn` to `info`.